sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs

```diff
@@ -2,147 +2,183 @@
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "latest";
 
-            Assert.AreEqual(_tagName, properties.Name);
+            // Act
+            TagProperties properties = await client.GetTagPropertiesAsync(tag);
+
+            // Assert
+            Assert.AreEqual(tag, properties.Name);
             Assert.AreEqual(_repositoryName, properties.Repository);
             Assert.AreEqual(new Uri(TestEnvironment.Endpoint).Host, properties.Registry);
         }
 
         [RecordedTest, NonParallelizable]
-        public async Task CanSetTagProperties([Values(true, false)] bool canList,
-                                              [Values(true, false)] bool canRead,
-                                              [Values(true, false)] bool canWrite,
-                                              [Values(true, false)] bool canDelete)
+        public async Task CanSetTagProperties()
         {
-            await _client.SetTagPropertiesAsync(
-                _tagName,
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "latest";
+            TagProperties tagProperties = await client.GetTagPropertiesAsync(tag);
+            ContentProperties originalContentProperties = tagProperties.ModifiableProperties;
+
+            // Act
+            await client.SetTagPropertiesAsync(
+                tag,
                 new ContentProperties()
                 {
-                    CanList = canList,
-                    CanRead = canRead,
-                    CanWrite = canWrite,
-                    CanDelete = canDelete
+                    CanList = false,
+                    CanRead = false,
+                    CanWrite = false,
+                    CanDelete = false
                 });
 
-            TagProperties properties = await _client.GetTagPropertiesAsync(_tagName);
+            // Assert
+            TagProperties properties = await client.GetTagPropertiesAsync(tag);
+
+            Assert.IsFalse(properties.ModifiableProperties.CanList);
+            Assert.IsFalse(properties.ModifiableProperties.CanRead);
+            Assert.IsFalse(properties.ModifiableProperties.CanWrite);
+            Assert.IsFalse(properties.ModifiableProperties.CanDelete);
+
+            // Cleanup
+            await client.SetTagPropertiesAsync(tag, originalContentProperties);
+        }
+
+        [RecordedTest, NonParallelizable]
+        public async Task CanDeleteTag()
+        {
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "test-delete";
+            await ImportImage(tag);
+
+            // Act
+            await client.DeleteTagAsync(tag);
+
+            // Assert
+
+            // The delete takes some time, so if we call GetTagProperties() without a delay, we get a 200 response.
+            await Task.Delay(5000);
```

Comment:
I would avoid this pattern because it forces us to always wait 5 sec. Instead it might be better to poll and until the key is not there anymore.

Similar to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Keys/tests/KeysTestBase.cs#L267-L288

Reply:
I'm investigating this behavior with the service right now.  We think it shouldn't be happening, but I need to go deeper on why we're seeing this in the client.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/19699